Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,284 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ForgeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The ForgeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+namespace Microsoft.Forge.TreeWalker
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Microsoft.Forge.DataContracts;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+    /// <summary>
+    /// The ForgeSchemaValidator class implements the validation method that tests input schemas with custom rules from input.
+    /// </summary>
+    public static class ForgeSchemaValidator
+    {
+        /// <summary>
+        /// The GetLinkedJSchema method that creates the linked rules in JSchema.
+        /// </summary>
+        /// <param name="childRules">The rules to be included in the parent rules</param>
+        /// <param name="parentRules">The parent rules to absorb the child rules</param>
+        /// <param name="referenceUri">The address of childRules</param>
+        /// <returns>The result of schema combination. ErrorMessage would be set if it throw exceptions</returns>
+        public static JSchema GetLinkedJSchemaRules(string childRules, string parentRules, string referenceUri, out string errorMessage)
+        {
+            try
+            {
+                JSchemaPreloadedResolver resolver = new JSchemaPreloadedResolver();
+                resolver.Add(new Uri(referenceUri), childRules);
+                errorMessage = "";
+                return JSchema.Parse(parentRules, resolver);
+            }
+            catch(Exception e)
+            {
+                errorMessage = e.Message;
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// The validate task that validate the input schema with custom rules in string.
+        /// </summary>
+        /// <param name="schema">The schema to be validated</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchema(ForgeTree schema, string rules, out List<string> errorList)
```

Comment:
>string rules [](http://example.com/codeflow?start=60&length=12)

I liked the idea to use object rules and then we can figure out if it is a string or JSchema. This will halve the number of methods required, which is a bonus.